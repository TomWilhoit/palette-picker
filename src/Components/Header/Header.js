import React, { Component } from "react";

export class Header extends Component {

  handleClick = () => {
    this.props.toggleInfo();
  }

  render() {
    return (
      <div className="header">
        <h1>Palette Picker</h1>
        <div className="info-btn" onClick={this.handleClick}>
          <i className="fa fa-info" aria-hidden="true"></i>
        </div>
      </div>
    )
  }
}