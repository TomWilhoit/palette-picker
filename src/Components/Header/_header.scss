.header {
  position: relative;
}

h1 {
  color: $title;
  margin: 0;
  padding: 0;
  text-align: center;
  size: 450%;
}

.info-btn {
  @include icon-btn-hover;
  border: 2px solid $border;
  border-radius: 25px;
  color: #415399;
  height: 24px;
  position: absolute;
  left: 65%;
  bottom: 72%;
  width: 24px;
  i {
    @include icon-down;
    font-size: 12px;
  }
  &:hover {
    background-color: $console-hover;
  }
}