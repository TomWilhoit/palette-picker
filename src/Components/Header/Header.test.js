import React from "react";
import { Header } from "./Header";
import { shallow } from "enzyme";

describe("Header", () => {
  let wrapper;
  
  beforeEach(() => {
    wrapper = shallow(<Header />);
  })

  it("should match the snapshot with all data passed in", () => {
    expect(wrapper).toMatchSnapshot();
  })

  it("should return an element", () => {
    const expected = (
      <h1>
        Palette Picker
      </h1>
    );
    let result = Header();
    expect(result).toEqual(expected);
  })
})
