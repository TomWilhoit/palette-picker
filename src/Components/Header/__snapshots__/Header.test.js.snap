// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should match the snapshot with all data passed in 1`] = `
<h1>
  Palette Picker
</h1>
`;
